onnxruntime/core/framework/tensor_external_data_info.cc

```diff
@@ -25,28 +46,110 @@ Status ExternalDataInfo::Create(const RepeatedPtrField<StringStringEntryProto>&
     if (stringmap.key() == "location" && !stringmap.value().empty()) {
       out->rel_path_ = ToWideString(stringmap.value());
     } else if (stringmap.key() == "offset" && !stringmap.value().empty()) {
-      char* end;
-#ifdef _WIN32
-      out->offset_ = _strtoi64(stringmap.value().c_str(), &end, 10);
-#else
-      out->offset_ = OrtStrToPtrDiff(stringmap.value().c_str(), &end);
-#endif
-      if (end != stringmap.value().c_str() + stringmap.value().length())
-        return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "parsing ", stringmap.value(), " failed");
+      ORT_RETURN_IF_ERROR(str_to_int(stringmap.value(), out->offset_));
     } else if (stringmap.key() == "length" && !stringmap.value().empty()) {
       char* end;
-      out->length_ = static_cast<size_t>(OrtStrToPtrDiff(stringmap.value().c_str(), &end));
+      out->length_ = narrow<size_t>(OrtStrToPtrDiff(stringmap.value().c_str(), &end));
       if (end != stringmap.value().c_str() + stringmap.value().length())
         return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "parsing ", stringmap.value(), " failed");
     } else if (stringmap.key() == "checksum" && !stringmap.value().empty()) {
       out->checksum_ = stringmap.value();
+    } else if (stringmap.key().find("prepacked", 0) == 0) {
+      // Starts with 'prepacked', each has its own key.
+      // Each prepacked entry may have multiple blobs with the same key
+      // we output them with the same key
+      // format = key|offset;length;checksum[|offset;length;checksum]
+      // We are ignoring invalid entries (should not be any), and rely
+      // on in memory pre-packs regenerated in this case.
+      // users can over-write this file with the correct pre-packed info.
+      const std::string& prepacked = stringmap.value();
+      if (!prepacked.empty()) {
+        auto split_fields = utils::SplitString(prepacked, "|", false);
+        if (split_fields.size() > 1) {
+          const std::string key{split_fields[0]};
+          auto& blob_infos = prepacked_infos[key];
+          for (size_t f = 1; f < split_fields.size(); ++f) {
+            const auto& blob = split_fields[f];
+            auto blob_fields = utils::SplitString(blob, ";", false);
+            if (blob_fields.size() == 3) {
```

Comment:
Here blob_fields.size() != 3 is ignored. Shall we return failure in that case.

Reply:
I have been pondering on this, and decided to be forgiving. After all, if a pre-packed blob is not loaded, then it is going to re-generated.